src/compose/ComposeBox.js

```diff
@@ -432,49 +502,30 @@ class ComposeBoxInner extends PureComponent<Props, State> {
     dispatch(sendTypingStop(destinationNarrow));
   };
 
-  inputMarginPadding = {
-    paddingHorizontal: 8,
-    paddingVertical: Platform.select({
-      ios: 8,
-      android: 2,
-    }),
-  };
+  getValidationErrors = (): $ReadOnlyArray<ValidationError> => {
+    const { mandatoryTopics } = this.props;
+    const destinationNarrow = this.getDestinationNarrow();
+    const { message } = this.state;
+
+    const result = [];
 
-  styles = {
-    wrapper: {
-      flexShrink: 1,
-      maxHeight: '60%',
-    },
-    autocompleteWrapper: {
-      position: 'absolute',
-      bottom: 0,
-      width: '100%',
-    },
-    composeBox: {
-      flexDirection: 'row',
-      alignItems: 'flex-end',
-      flexShrink: 1,
-    },
-    composeText: {
-      flex: 1,
-      paddingVertical: 8,
-    },
-    composeSendButton: {
-      padding: 8,
-    },
-    topicInput: {
-      borderWidth: 0,
-      borderRadius: 5,
-      marginBottom: 8,
-      ...this.inputMarginPadding,
-    },
-    composeTextInput: {
-      borderWidth: 0,
-      borderRadius: 5,
-      fontSize: 15,
-      flexShrink: 1,
-      ...this.inputMarginPadding,
-    },
+    if (
+      isTopicNarrow(destinationNarrow)
+      && topicOfNarrow(destinationNarrow) === apiConstants.NO_TOPIC_TOPIC
+      && mandatoryTopics
```

Comment:
nit: let's put `mandatoryTopics` first -- that sets the context for what the rest of the condition is about, and it also can short-circuit the rest of the logic early (though the latter doesn't seriously matter because this is all pretty cheap.)

Reply:
Ah I see, this is verbatim from the old code at `handleSend`. Shrug.

You can turn it into a code move by squashing that next commit

    ComposeBox [nfc]: Deduplicate expression for mandatory-topic validation

into this one. The diff gets just a bit bigger, but the real complexity of the diff is actually smaller, I think (even just comparing the squashed commit to this first one, and more so if comparing to both the commits), because this code gets moved instead of introduced.